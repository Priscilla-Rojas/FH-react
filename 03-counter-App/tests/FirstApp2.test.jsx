import { render, screen } from "@testing-library/react"
import FirstApp from "../src/firstApp"

describe('Pruebas en <FirstApp/>', () => {
  
  const title= 'I am title';
  const subTitle= 'I am subTitle';

  test('Debe Hacer match con el Snapshot', () => { 

    const { container } =  render( <FirstApp title={title}/> )
    expect ( container ).toMatchSnapshot();

  })
  
  test('debe mostrar el mensaje "I am title"', () => {
    
    render( <FirstApp title={title}/> )
    expect( screen.getByText( title )).toBeTruthy(); 
    // screen.debug();

  })

  test('debe mostrar el titulo en un H1', () => {
    
    render( <FirstApp title={title}/> )
    expect( screen.getByRole( 'heading', { leavel: 1 } ).innerHTML).toContain(title);

  })

  test('debe mostrar el subtitulo enviado por props', () => {

    
    render( 
      <FirstApp title={title} subtitle={subTitle} /> 
    )
    expect( screen.getAllByText(subTitle).length).toBe(2); 
  })
})
