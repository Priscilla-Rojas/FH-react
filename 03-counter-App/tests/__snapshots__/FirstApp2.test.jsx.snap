// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> Debe Hacer match con el Snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    I am title
  </h1>
  <p>
    no hay subtitulo
  </p>
  <p>
    no hay subtitulo
  </p>
  <p>
    Priscilla Rojas
  </p>
</div>
`;
